File: Stack.java
import java.util.Scanner;

// Node class
class Node {
    int data;
    Node next;

    Node(int value) {
        data = value;
        next = null;
    }
}

// Stack class
class Stack {
    private Node first;
    private int count;

    public Stack() {
        first = null;
        count = 0;
    }

    public void push(int no) {    // InsertLast
        Node newn = new Node(no);

        if (first == null) {
            first = newn;
        } else {
            Node temp = first;
            while (temp.next != null) {
                temp = temp.next;
            }
            temp.next = newn;
        }
        count++;
    }

    public int pop() {    // DeleteLast
        int value = 0;

        if (count == 0) {
            System.out.println("Stack is empty");
            return -1;
        } else if (count == 1) {
            value = first.data;
            first = null;
        } else {
            Node temp = first;
            while (temp.next.next != null) {
                temp = temp.next;
            }
            value = temp.next.data;
            temp.next = null;
        }

        count--;
        return value;
    }

    public void display() {
        if (first == null) {
            System.out.println("Nothing to display as Stack is empty");
            return;
        }

        System.out.println("Elements of stack are : ");
        Node temp = first;
        while (temp != null) {
            System.out.println(temp.data);
            temp = temp.next;
        }
    }

    public int count() {
        return count;
    }
}

// Main class
public class StackDemo {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int iChoice = 1;
        int iValue = 0;
        int iRet = 0;

        Stack obj = new Stack();

        System.out.println("-----------------------------------------------------");
        System.out.println("Dynamic Implementation of Stack in Java");
        System.out.println("-----------------------------------------------------");

        while (iChoice != 0) {
            System.out.println("-----------------------------------------------------");
            System.out.println("1 : Push element into stack");
            System.out.println("2 : Pop element from the stack");
            System.out.println("3 : Display elements from stack");
            System.out.println("4 : Count number of elements from stack");
            System.out.println("0 : Terminate the application");
            System.out.println("-----------------------------------------------------");

            System.out.print("Please enter the option : ");
            iChoice = sc.nextInt();

            switch (iChoice) {
                case 1:
                    System.out.print("Enter the element that you want to push: ");
                    iValue = sc.nextInt();
                    obj.push(iValue);
                    break;

                case 2:
                    iRet = obj.pop();
                    if (iRet != -1) {
                        System.out.println("Popped element from stack is : " + iRet);
                    }
                    break;

                case 3:
                    obj.display();
                    break;

                case 4:
                    iRet = obj.count();
                    System.out.println("Number of elements are : " + iRet);
                    break;

                case 0:
                    System.out.println("Thank you for using the application");
                    break;

                default:
                    System.out.println("Please enter valid option");
                    break;
            }
        }

        sc.close();
    }
}

File: Stack.c
#include <stdio.h>
#include <stdlib.h>

// Node structure
typedef struct node
{
    int data;
    struct node *next;
} NODE, *PNODE;

// Stack structure
typedef struct stack
{
    PNODE First;
    int iCount;
} STACK;

// Function prototypes
void Initialize(STACK *obj);
void Push(STACK *obj, int no);
int Pop(STACK *obj);
void Display(STACK *obj);
int Count(STACK *obj);

void Initialize(STACK *obj)
{
    obj->First = NULL;
    obj->iCount = 0;
}

void Push(STACK *obj, int no) // InsertLast
{
    PNODE newn = (PNODE)malloc(sizeof(NODE));
    newn->data = no;
    newn->next = NULL;

    if (obj->First == NULL)
    {
        obj->First = newn;
    }
    else
    {
        PNODE temp = obj->First;
        while (temp->next != NULL)
        {
            temp = temp->next;
        }
        temp->next = newn;
    }
    obj->iCount++;
}

int Pop(STACK *obj) // DeleteLast
{
    int Value = 0;

    if (obj->iCount == 0)
    {
        printf("Stack is empty\n");
        return -1;
    }
    else if (obj->iCount == 1)
    {
        Value = obj->First->data;
        free(obj->First);
        obj->First = NULL;
    }
    else
    {
        PNODE temp = obj->First;

        while (temp->next->next != NULL)
        {
            temp = temp->next;
        }

        Value = temp->next->data;
        free(temp->next);
        temp->next = NULL;
    }

    obj->iCount--;

    return Value;
}

void Display(STACK *obj)
{
    if (obj->First == NULL)
    {
        printf("Nothing to display as Stack is empty\n");
        return;
    }

    printf("Elements of stack are : \n");

    PNODE temp = obj->First;

    while (temp != NULL)
    {
        printf("%d\n", temp->data);
        temp = temp->next;
    }
}

int Count(STACK *obj)
{
    return obj->iCount;
}

// Main function
int main()
{
    int iChoice = 1;
    int iValue = 0;
    int iRet = 0;

    STACK obj;
    Initialize(&obj);

    printf("-----------------------------------------------------\n");
    printf("Dynamic Implementation of Stack in C\n");
    printf("-----------------------------------------------------\n");

    while (iChoice != 0)
    {
        printf("-----------------------------------------------------\n");
        printf("1 : Push element into stack\n");
        printf("2 : Pop element from the stack\n");
        printf("3 : Display elements from stack\n");
        printf("4 : Count number of elements from stack\n");
        printf("0 : Terminate the application\n");
        printf("-----------------------------------------------------\n");

        printf("Please enter the option : \n");
        scanf("%d", &iChoice);

        switch (iChoice)
        {
        case 1:
            printf("Enter the element that you want to push\n");
            scanf("%d", &iValue);
            Push(&obj, iValue);
            break;

        case 2:
            iRet = Pop(&obj);
            if (iRet != -1)
            {
                printf("Poped element from stack is : %d\n", iRet);
            }
            break;

        case 3:
            Display(&obj);
            break;

        case 4:
            iRet = Count(&obj);
            printf("Number of elements are : %d\n", iRet);
            break;

        case 0:
            printf("Thank you for using the application\n");
            break;

        default:
            printf("Please enter valid option\n");
            break;
        } // End of switch
    }     // End of while

    return 0;
}
